### PR TITLE
ci: Clear unneeded runner dirs for stake pool job

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

The stake pool CI job is failing with "no space left on device".

#### Solution

Blast some unused directories and see if that helps!

### CODE REVIEW

File: ci/warning/purge-ubuntu-runner.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+# !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+# WARNING: THIS IS NOT TO BE USED LOCALLY, BUT ONLY IN GITHUB ACTIONS TO CLEAR
+# DISK SPACE ON UBUNTU. IT IS NOT MARKED AS EXECUTABLE BY DEFAULT FOR SAFETY.
+# !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+sudo rm -rf /usr/share/dotnet /usr/local/lib/android /opt/ghc
```
Comment (t-nelson) [inline ci/warning/purge-ubuntu-runner.sh]: is this necessary with `dotnet-sdk-* mono-complete ghc-*` purges below?  any idea where the android stuff comes from?

Comment (joncinque) [reply ci/warning/purge-ubuntu-runner.sh]: Good point, let's see what happens without this line. `/usr/local/lib/android` isn't installed with `android-sdk`, so honestly I have no idea. Let's roll the dice!

**Comment by t-nelson** [comment]
maybe another source of unneeded disk consumption https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#cached-docker-images

**Comment by joncinque** [comment]
> maybe another source of unneeded disk consumption https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#cached-docker-images

ah nice find, done in [4ede450](https://github.com/solana-labs/solana-program-library/pull/4505/commits/4ede450329b8adfbd9cabe5ddae4d42c1a0f56ba)

**Comment by t-nelson** [comment]
where do these put us with free space, sufficient?

File: .github/workflows/pull-request-stake-pool.yml [hunk 2]
```diff
@@ -6,19 +6,24 @@ on:
...
     - 'ci/*-version.sh'
+    - '.github/workflows/pull-request-stake-pool.yml'
 
 jobs:
   cargo-test-sbf:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
 
+      - name: Clear dirs for more space
```
Comment (t-nelson) [inline .github/workflows/pull-request-stake-pool.yml:24]:
```suggestion
      - name: Remove unneeded packages for more space
```

nit

**Comment by yihau** [comment]
not sure is it help, but provide the pre-install software list here:
https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md

if we don't need android stuff, we can delete them and get ~10G back 👻

**Comment by yihau** [comment]
I also found it installed lots of unnecessary tools in `/usr/local/bin`

<details>
  <summary>Click me</summary>
ls -alh

total 1.2G
drwxrwxrwx  2 root   root   4.0K May 17 22:40 .
drwxr-xr-x 19 root   root   4.0K May 17 22:16 ..
-rwxr-xr-x  1 root   root   3.3K May 17 22:32 activate-global-python-argcomplete
-rwxrwxrwx  1 root   root    20M May 16 09:19 aliyun
lrwxrwxrwx  1 root   root     37 May 17 21:26 aws -> /usr/local/aws-cli/v2/current/bin/aws
lrwxrwxrwx  1 root   root     47 May 17 21:26 aws_completer -> /usr/local/aws-cli/v2/current/bin/aws_completer
-rwxrwxrwx  1 runner docker  30M Apr 20 22:03 azcopy
lrwxrwxrwx  1 root   root     21 May 17 21:23 azcopy10 -> /usr/local/bin/azcopy
lrwxrwxrwx  1 root   root     47 May 17 22:01 bazel -> ../lib/node_modules/@bazel/bazelisk/bazelisk.js
lrwxrwxrwx  1 root   root     47 May 17 22:01 bazelisk -> ../lib/node_modules/@bazel/bazelisk/bazelisk.js
lrwxrwxrwx  1 root   root     24 May 17 21:[5](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:6)7 bcp -> /opt/mssql-tools/bin/bcp
-rwxrwxrwx  1 root   root    5[6](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:7)M May 1[7](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:8) 21:25 bicep
-rwxr-xr-x  1 root   root    556 May 17 22:05 bin-proxy
-rwxrwxrwx  1 root   root    14M Apr  4 19:10 ccmake
-rwxrwxrwx  1 root   root    14M Apr  4 19:10 cmake
-rwxrwxrwx  1 root   root    29M Apr  4 19:10 cmake-gui
-rwxr-xr-x  1 root   root    560 May 17 22:05 commander
lrwxrwxrwx  1 root   root     45 May 17 21:5[8](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:9) corepack -> ../lib/node_modules/corepack/dist/corepack.js
-rwxrwxrwx  1 root   root    14M Apr  4 1[9](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:10):[10](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:11) cpack
-rwxrwxrwx  1 root   root    15M Apr  4 19:10 ctest
-rwxrwxrwx  1 root   root    13M May 17 21:13 docker-compose
-rwxr-xr-x  1 root   root    542 May 17 22:05 dotenv
-rwxr-xr-x  1 root   root    554 May 17 22:05 fastlane
lrwxrwxrwx  1 root   root     35 May 17 22:00 grunt -> ../lib/node_modules/grunt/bin/grunt
lrwxrwxrwx  1 root   root     36 May 17 22:00 gulp -> ../lib/node_modules/gulp/bin/gulp.js
-rwxrwxrwx  1 root   root    49M May 17 21:52 helm
lrwxrwxrwx  1 root   root     32 May 17 21:50 heroku -> /usr/local/lib/heroku/bin/heroku
-rwxr-xr-x  1 root   root    566 May 17 22:05 httpclient
-rwxrwxrwx  1 runner docker 9.4M Mar  5  2020 hub
-rwxr-xr-x  1 root   root    560 May 17 22:05 jmespath.rb
-rwxrwxrwx  1 root   root   6.7M May 17 21:52 kind
-rwxrwxrwx  1 root   root    47M May 17 21:52 kubectl
-rwxrwxrwx  1 root   root    15M May 17 21:52 kustomize
-rwxrwxrwx  1 root   root    14K May 17 21:52 lein
lrwxrwxrwx  1 root   root     37 May 17 22:00 lerna -> ../lib/node_modules/lerna/dist/cli.js
-rwxrwxrwx  1 root   root    81M May 17 21:52 minikube
lrwxrwxrwx  1 root   root     27 May 17 22:00 n -> ../lib/node_modules/n/bin/n
lrwxrwxrwx  1 root   root     43 May 17 22:00 netlify -> ../lib/node_modules/netlify-cli/bin/run.mjs
lrwxrwxrwx  1 root   root     40 May 17 22:00 newman -> ../lib/node_modules/newman/bin/newman.js
-rwxrwxrwx  1 root   root    87M May 17 21:58 node
lrwxrwxrwx  1 root   root     21 May 17 22:00 now -> /usr/local/bin/vercel
lrwxrwxrwx  1 root   root     38 May 17 21:58 npm -> ../lib/node_modules/npm/bin/npm-cli.js
lrwxrwxrwx  1 root   root     38 May 17 21:58 npx -> ../lib/node_modules/npm/bin/npx-cli.js
lrwxrwxrwx  1 root   root     43 May 17 22:00 ntl -> ../lib/node_modules/netlify-cli/bin/run.mjs
-rwxrwxrwx  1 root   root   125M May  2 22:14 oc
-rwxr-xr-x  1 runner docker 8.8M Mar 20 10:12 oras
-rwxr-xr-x  1 root   root   195M May  4 19:17 packer
lrwxrwxrwx  1 root   root     37 May 17 22:00 parcel -> ../lib/node_modules/parcel/lib/bin.js
-rwxr-xr-x  1 root   root   3.1M Feb 27 13:05 phpunit
-rwxr-xr-x  1 root   root    206 May 17 22:32 pipx
-rwxr-xr-x  1 runner docker  80M May 15 19:05 pulumi
-rwxr-xr-x  1 runner docker  245 May 15 19:08 pulumi-analyzer-policy
-rwxr-xr-x  1 runner docker 1.9K May 15 19:08 pulumi-analyzer-policy-python
-rwxr-xr-x  1 runner docker  23M Feb 13 17:44 pulumi-language-dotnet
-rwxr-xr-x  1 runner docker  66M May 15 19:05 pulumi-language-go
-rwxr-xr-x  1 runner docker  24M Apr 18 19:34 pulumi-language-java
-rwxr-xr-x  1 runner docker  24M May 15 19:05 pulumi-language-nodejs
-rwxr-xr-x  1 runner docker  24M May 15 19:05 pulumi-language-python
-rwxr-xr-x  1 runner docker  [11](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:12)K May 15 19:08 pulumi-language-python-exec
-rwxr-xr-x  1 runner docker  22M Apr [12](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:13) 17:09 pulumi-language-yaml
-rwxr-xr-x  1 runner docker  238 May [15](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:16) 19:08 pulumi-resource-pulumi-nodejs
-rwxr-xr-x  1 runner docker 1.9K May 15 19:08 pulumi-resource-pulumi-python
-rwxr-xr-x  1 runner docker 6.6M Jul 25  2022 pulumi-watch
-rwxr-xr-x  1 root   root   2.6K May [17](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:18) 22:32 python-argcomplete-check-easy-install-script
-rwxr-xr-x  1 root   root   1.1M May 17 22:34 pzstd
-rwxr-xr-x  1 root   root   2.0K May 17 22:32 register-python-argcomplete
-rwxr-xr-x  1 root   root    540 May 17 22:05 rougify
lrwxrwxrwx  1 root   root     38 May 17 21:26 sam -> /usr/local/aws-sam-cli/current/bin/sam
lrwxrwxrwx  1 root   root     58 May 17 21:26 session-manager-plugin -> /usr/local/sessionmanagerplugin/bin/session-manager-plugin
lrwxrwxrwx  1 root   root     27 May 17 21:57 sqlcmd -> /opt/mssql-tools/bin/sqlcmd
lrwxrwxrwx  1 root   root     32 May 17 21:57 sqlpackage -> /usr/local/sqlpackage/sqlpackage
-rwxrwxrwx  1 root   root    54M May 17 21:50 stack
lrwxrwxrwx  1 root   root     30 May 17 21:29 swift -> /usr/share/swift/usr/bin/swift
lrwxrwxrwx  1 root   root     31 May 17 21:29 swiftc -> /usr/share/swift/usr/bin/swiftc
-rwxr-xr-x  1 root   root    608 May 17 22:05 terminal-notifier
-rwxr-xr-x  1 root   root    62M Apr 26 17:54 terraform
lrwxrwxrwx  1 root   root     38 May 17 22:00 tsc -> ../lib/node_modules/typescript/bin/tsc
lrwxrwxrwx  1 root   root     43 May 17 22:00 tsserver -> ../lib/node_modules/typescript/bin/tsserver
lrwxrwxrwx  1 root   root     [19](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:20) May 17 22:34 unzstd -> /usr/local/bin/zstd
-rwxr-xr-x  1 root   root    [21](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:22)9 May 17 [22](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:23):32 userpath
lrwxrwxrwx  1 root   root     40 May 17 22:00 vc -> ../lib/node_modules/vercel/dist/index.js
lrwxrwxrwx  1 root   root     [28](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:29) May 17 22:17 vcpkg -> /usr/local/share/vcpkg/vcpkg
lrwxrwxrwx  1 root   root     40 May 17 22:00 vercel -> ../lib/node_modules/vercel/dist/index.js
lrwxrwxrwx  1 root   root     42 May 17 22:00 webpack -> ../lib/node_modules/webpack/bin/webpack.js
lrwxrwxrwx  1 root   root     42 May 17 22:00 webpack-cli -> ../lib/node_modules/webpack-cli/bin/cli.js
-rwxr-xr-x  1 root   root    546 May 17 22:05 ww
-rwxr-xr-x  1 root   root    560 May 17 22:05 xcodeproj
-rwxr-xr-x  1 root   root    554 May 17 22:05 xcpretty
-rwxr-xr-x  1 root   root    656 May 17 22:05 xcpretty-travis-formatter
lrwxrwxrwx  1 root   root     36 May 17 22:00 yarn -> ../lib/node_modules/yarn/bin/yarn.js
lrwxrwxrwx  1 root   root     36 May 17 22:00 yarnpkg -> ../lib/node_modules/yarn/bin/yarn.js
-rwxr-xr-x  1 root   root   1.2M May 17 22:[34](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:35) zstd
lrwxrwxrwx  1 root   root     19 May 17 22:34 zstdcat -> /usr/local/bin/zstd
-rwxr-xr-x  1 root   root   3.8K May 17 22:34 zstdgrep
-rwxr-xr-x  1 root   root    1[97](https://github.com/yihau/playground/actions/runs/5208562353/jobs/9397297322#step:2:98) May 17 22:34 zstdless
lrwxrwxrwx  1 root   root     19 May 17 22:34 zstdmt -> /usr/local/bin/zstd

**Comment by joncinque** [comment]
Merging this now to unblock CI, we can always refine later

**Pull request merged**